vertx-consul-client/src/test/java/io/vertx/ext/consul/suite/Watches.java

```diff
@@ -0,0 +1,235 @@
+    assertEquals(outs.get(2), "v1");
+    assertEquals(outs.get(3), "v2");
+    assertEquals(outs.get(4), "Not Found");
+  }
+
+  @Test
+  public void watchExistingKey() throws InterruptedException {
+    List<String> outs = new ArrayList<>();
+
+    assertTrue(getAsync(h -> writeClient.putValue("foo/bar/w2", "v1", h)));
+
+    Watch<KeyValue> watch = Watch.key("foo/bar/w2", vertx, readClientOptions)
+      .setHandler(kv -> {
+        if (kv.succeeded()) {
+          outs.add(kv.result().getValue());
+        } else {
+          outs.add(kv.cause().getMessage());
+        }
+      })
+      .start();
+    sleep(vertx, 1000);
+
+    assertTrue(getAsync(h -> writeClient.putValue("foo/bar/w2", "v2", h)));
+    sleep(vertx, 1000);
+    runAsync(h -> writeClient.deleteValue("foo/bar/w2", h));
+    sleep(vertx, 1000);
+    watch.stop();
+
+    assertEquals(outs.get(0), "v1");
+    assertEquals(outs.get(1), "v2");
+    assertEquals(outs.get(2), "Not Found");
+  }
+
+  @Test
+  public void watchServices() {
+
+    List<String> outs = new ArrayList<>();
+
+    ServiceOptions service = new ServiceOptions()
+      .setId(randomAlphaString(10))
+      .setName(randomAlphaString(10));
+
+    Watch<ServiceList> watch = Watch.services(vertx, readClientOptions)
+      .setHandler(list -> {
+        if (list.succeeded()) {
+          outs.add(list.result().getList()
+            .stream().map(Service::getName).filter(s -> s.equals(service.getName()))
+            .findFirst().orElse(""));
+        } else {
+          outs.add(list.cause().getMessage());
+        }
+      })
+      .start();
+    sleep(vertx, 1000);
+
+    runAsync(h -> writeClient.registerService(service, h));
+    sleep(vertx, 1000);
+    watch.stop();
+
+    assertEquals(outs.get(0), "");
+    assertEquals(outs.get(1), service.getName());
+
+    runAsync(h -> writeClient.deregisterService(service.getId(), h));
+  }
+
+  @Test
+  public void watchService() {
+
+    List<String> outs = new ArrayList<>();
+
+    ServiceOptions service = new ServiceOptions()
+      .setCheckOptions(new CheckOptions()
+        .setStatus(CheckStatus.PASSING)
+        .setTtl("2s")
+        .setName(randomAlphaString(10)))
+      .setId(randomAlphaString(10))
+      .setName(randomAlphaString(10));
+
+    Watch<ServiceEntryList> watch = Watch.service(service.getName(), vertx, readClientOptions)
+      .setHandler(list -> {
+        if (list.succeeded()) {
+          outs.add(list.result().getList()
+            .stream().filter(s -> s.getService().getName().equals(service.getName()))
+            .map(e -> e.getService().getName() + "/" + e.getChecks().stream()
+              .filter(c -> c.getId().equals("service:" + service.getId()))
+              .map(c -> c.getStatus().name()).findFirst().orElse(""))
+            .findFirst().orElse(""));
+        } else {
+          outs.add(list.cause().getMessage());
+        }
+      })
+      .start();
+    sleep(vertx, 1000);
```

Comment:
Can't we use either vert.x unit of Awaitability to avoid these hard-coded time. It will likely be an issue in slow CI.